priam/src/main/java/com/netflix/priam/tuner/PropertiesFileTuner.java

```diff
@@ -1,87 +1,55 @@
 package com.netflix.priam.tuner;
 
-import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Splitter;
+import com.google.inject.Inject;
+import com.netflix.priam.config.IConfiguration;
 import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.configuration2.ex.ConfigurationException;
-import org.apache.commons.lang3.text.StrSubstitutor;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.text.StringSubstitutor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.inject.Inject;
-import com.netflix.priam.config.IConfiguration;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
 
 /**
  * Support tuning standard .properties files
  * <p>
  */
 public class PropertiesFileTuner
 {
-    private static final Logger logger = LoggerFactory.getLogger(JVMOptionsTuner.class);
+    private static final Logger logger = LoggerFactory.getLogger(PropertiesFileTuner.class);
     protected final IConfiguration config;
-    protected final String propertyPrefix;
 
     @Inject
-    public PropertiesFileTuner(IConfiguration config, String prefix) {
+    public PropertiesFileTuner(IConfiguration config)
+    {
         this.config = config;
-        this.propertyPrefix = prefix;
     }
 
     @SuppressWarnings("unchecked")
-    public void updateAndSaveProperties(String configPath) throws IOException, ConfigurationException
+    public void updateAndSaveProperties(String propertyFile) throws IOException, ConfigurationException
     {
-        File propertiesFile = new File(configPath);
         try {
-            if (propertiesFile.exists() && !propertiesFile.canWrite()) {
-                throw new IOException("Can't write and therefore cannot tune" + configPath);
-            }
-
             PropertiesConfiguration properties = new PropertiesConfiguration();
-            properties.getLayout().load(properties, new FileReader(propertiesFile.getPath()));
-
-            Set<String> keys = new HashSet<>();
-            properties.getKeys().forEachRemaining(keys::add);
-
-            Map<String, String> overridenProperties = new HashMap<>();
-            String propertyOverrides = config.getProperty("propertyOverrides." + propertyPrefix, null);
-
-            if (propertyOverrides != null) {
+            properties.getLayout().load(properties, new FileReader(propertyFile));
```

Comment:
What happens if the propertyFile does not exist? Is that handled by the FileReader in some way.

Reply:
Never mind it seems it is being handled in the exception handler below.